src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -195,6 +201,32 @@ public class CpsScmFlowDefinitionTest {
         r.assertLogContains("version one", b);
     }
 
+    @Test public void lightweight_brach_parametrised() throws Exception {
+        LoggerRule lr = new LoggerRule();
+        lr.record(GitSCMFileSystem.class.getName(), Level.ALL).capture(4024);
+
+        sampleRepo.init();
+        sampleRepo.git("checkout","-b","master2");
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.git("commit", "--message=init");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("BRANCH","","")));
+        GitStep step = new GitStep(sampleRepo.toString());
+        step.setBranch("${BRANCH}");
+
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(step.createSCM(), "flow.groovy");
```

Comment:
I think clearer to construct a `GitSCM` directly?

Reply:
actually I couldnt get my head around that API how to easily create it. But i can change it somehow